Do not link with a C++ standard library nor with zlib; tagtool does not use them.
id3lib uses them and, of course, links with them.
--- configure.ac.orig	2007-02-26 16:12:54.000000000 -0600
+++ configure.ac	2023-09-18 01:57:06.000000000 -0500
@@ -64,19 +64,10 @@
 		[enable_mp3=no;  disable_mp3_reason="(missing id3lib headers)"])
 fi;
 if test "$enable_mp3" = "yes"; then
-	dnl Fix missing link dependencies in libid3.so  (some distros have fixed 
-	dnl this in their own libid3 packages, others haven't...)
-	SAVE_LDFLAGS=$LDFLAGS
-	LDFLAGS="-lstdc++ -lz $LDFLAGS"
-
 	AC_CHECK_LIB(id3, main,,
 		[enable_mp3=no;  disable_mp3_reason="(missing id3 library)"])
 	AC_CHECK_LIB(id3, ID3FrameInfo_LongName,, 
 		[AC_DEFINE(LIBID3_MISSING_ID3FRAMEINFO, 1, [Define to 1 if the ID3FrameInfo_* functions are missing from libid3])])
-
-	if test "$enable_mp3" = "no"; then
-		LDFLAGS=$SAVE_LDFLAGS
-	fi;
 fi;
 
 if test "$enable_vorbis" = "yes"; then
